common/src/main/java/tachyon/util/network/NetworkAddressUtils.java

```diff
@@ -48,6 +50,84 @@
+   *
+   * @param conf Tachyon configuration used to look up the host resolution timeout
+   * @return the user specified hostname ($TACHYON_MASTER_BIND_HOST) for binding.
+   */
+  public static String getMasterHostName(TachyonConf conf) {
+    return conf.get(Constants.MASTER_HOSTNAME, getLocalHostName(conf));
+  }
+
+  /**
+   * Gets a binding hostname for master
+   *
+   * Host binding strategy on multihomed networking: 1) Environment variables via tachyon-env.sh or
+   * from OS settings 2) Default properties via tachyon-default.properties file 3) A reachable local
+   * host name for the host this JVM is running on
+   *
+   * @param conf Tachyon configuration used to look up the host resolution timeout
+   * @return the user specified hostname ($TACHYON_MASTER_BIND_HOST) for binding.
+   */
+  public static String getMasterBindHost(TachyonConf conf) {
+    return conf.get(Constants.MASTER_BIND_HOST, getLocalHostName(conf));
+  }
+
+  /**
+   * Gets a binding hostname for master web server
+   *
+   * @param conf Tachyon configuration used to look up the host resolution timeout
+   * @return the user specified hostname ($TACHYON_MASTER_WEB_BIND_HOST) for binding.
+   */
+  public static String getMasterWebBindHost(TachyonConf conf) {
+    return conf.get(Constants.MASTER_WEB_BIND_HOST, getLocalHostName(conf));
+  }
+
+  /**
+   * Helper method to get the {@link java.net.InetSocketAddress} connect hostname of the worker.
+   *
+   * @param conf the configuration of Tachyon
+   * @return the worker's connect hostname
+   */
+  public static String getWorkerConnectHost(TachyonConf conf) {
+    if (conf.get(Constants.WORKER_BIND_HOST, "localhost").equals("0.0.0.0")) {
+      return NetworkAddressUtils.getLocalHostName(conf);
+    }
+    return getWorkerBindHost(conf);
+  }
+
+  /**
+   * Gets a binding hostname for worker
+   *
+   * @param conf Tachyon configuration used to look up the host resolution timeout
+   * @return the user specified hostname ($TACHYON_WORKER_BIND_HOST) for binding.
+   */
+  public static String getWorkerBindHost(TachyonConf conf) {
+    return conf.get(Constants.WORKER_BIND_HOST, getLocalHostName(conf));
+  }
+
+  /**
+   * Gets a binding hostname for worker data server
+   *
+   * @param conf Tachyon configuration used to look up the host resolution timeout
+   * @return the user specified hostname ($TACHYON_WORKER_DATA_BIND_HOST) for binding.
+   */
+  public static String getWorkerDataBindHost(TachyonConf conf) {
+    return conf.get(Constants.WORKER_DATA_BIND_HOST, getLocalHostName(conf));
+  }
+
+  /**
+   * Gets a binding hostname for worker web server
+   *
+   * @param conf Tachyon configuration used to look up the host resolution timeout
+   * @return the user specified hostname ($TACHYON_WORKER_WEB_BIND_HOST) for binding.
+   */
+  public static String getWorkerWebBindHost(TachyonConf conf) {
+    return conf.get(Constants.WORKER_WEB_BIND_HOST, getLocalHostName(conf));
+  }
+
+  /**
```

Comment:
I know this PR doesn't change these methods, but it looks like `getLocalHostName` and `getLocalIpAddress` might not be thread-safe. If multiple threads call them at the same time, couldn't that cause problems if they both try to set the class variables, `sLocalHost` and `sLocalIp`?